pkg/config/analysis/analyzers/gateway/gateway.go

```diff
@@ -84,7 +84,13 @@ func (*IngressGatewayPortAnalyzer) analyzeGateway(r *resource.Instance, c analys
 				if svcSelector.Matches(podLabels) {
 					for _, port := range service.Ports {
 						if port.Protocol == "TCP" {
-							servicePorts[uint32(port.Port)] = true
+							// Because the Gateway's server port is the port on which the proxy should listen for incoming connections,
+							// the actual port associated with the service is the `TargetPort` that reaches the sidecar *workload instances*.
+							if tp := port.TargetPort.IntValue(); tp != 0 {
```

Comment:
I'm actually a little bit confused about the message description `The gateway refers to a port that is not exposed on the workload (pod selector %s; port %d)`, since the gateway is specifying which port to listen on.

Reply:
Not blocking this PR, and I think we may need to rephrase the message description as a followup.